lib/chai/core/assertions.js

```diff
@@ -2538,6 +2534,12 @@ function assertKeys(keys) {
     all = true;
   }
 
+  // Fix for PR #1384: Allow empty keys only if we are checking 'all'.
+  // 'any' keys requires at least one key to be checked against.
```

Comment:
```suggestion
  // Allow empty keys if we're checking `all` since you can assert it has _no keys_
  // For `any`, we need at least one key.
```

since the PR isn't relevant info anymore at this point (its a feature rather than a bug)